### PR TITLE
man: add working example to LogControl1 manpage

### PR DESCRIPTION
**Pull request by bluca**
Add fully working and documented example that can be copied and pasted

### CODE REVIEW

File: man/logcontrol-example.c [hunk 1]
```diff
@@ -0,0 +1,205 @@
...
+ *
+ * $ busctl --user get-property org.freedesktop.Example /org/freedesktop/LogControl1 org.freedesktop.LogControl1 SyslogIdentifier
+ *   s "example"
+ * $ busctl --user get-property org.freedesktop.Example /org/freedesktop/LogControl1 org.freedesktop.LogControl1 LogTarget
+ *   s "journal"
+ * $ busctl --user get-property org.freedesktop.Example /org/freedesktop/LogControl1 org.freedesktop.LogControl1 LogLevel
+ *   s "info"
+ * $ busctl --user set-property org.freedesktop.Example /org/freedesktop/LogControl1 org.freedesktop.LogControl1 LogLevel "s" debug
+ * $ busctl --user get-property org.freedesktop.Example /org/freedesktop/LogControl1 org.freedesktop.LogControl1 LogLevel
+ *   s "debug"
```
Comment (keszybz) [inline man/logcontrol-example.c]: Man pages should fit in 80–85 columns. Please break line break this.

File: man/logcontrol-example.c [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+    SD_BUS_VTABLE_PROPERTY_CONST),
+  SD_BUS_VTABLE_END
+};
+
+int main(int argc, char **argv) {
+  /* The bus should be relinquished once finished. The cleanup attribute
+   * allows to do it nice and cleanly whenever we exit the block. */
+  _cleanup_(sd_bus_flush_close_unrefp) sd_bus *bus = NULL;
+
+  /* Acquire a connection to the bus, letting the library figure out the details.
```
Comment (keszybz) [inline man/logcontrol-example.c]: The comments are too wide. C.f.
`MANWIDTH=80 build/man/man org.freedesktop.LogControl1 | rg '.{80,}'`

File: man/logcontrol-example.c [hunk 3]
```diff
@@ -0,0 +1,205 @@
...
+  LOG_TARGET_SYSLOG,
+  LOG_TARGET_CONSOLE,
+  _LOG_TARGET_MAX,
+} LogTarget;
+
+static const char* const log_target_table[_LOG_TARGET_MAX] = {
+  [LOG_TARGET_JOURNAL] = "journal",
+  [LOG_TARGET_KMSG] = "kmsg",
+  [LOG_TARGET_SYSLOG] = "syslog",
+  [LOG_TARGET_CONSOLE] = "console",
```
Comment (keszybz) [inline man/logcontrol-example.c:66]: Maybe align the strings in the usual fashion?
(Also in the next table.)

File: man/logcontrol-example.c [hunk 4]
```diff
@@ -0,0 +1,205 @@
...
+  SD_BUS_PROPERTY(
+    "SyslogIdentifier", "s",
+    property_get,
+    0,
+    SD_BUS_VTABLE_PROPERTY_CONST),
+  SD_BUS_VTABLE_END
+};
+
+int main(int argc, char **argv) {
+  /* The bus should be relinquished once finished. The cleanup attribute
```
Comment (keszybz) [inline man/logcontrol-example.c]: "once finished" implicitly refers to the subject of the sentence, i.e. the bus (effectively: "The bus should be relinquished once it is finished"), which is not what was meant. Maybe "before the program finishes"?

There's an adjective/adverb mismatch. Either "nice" and "clean" or "nicely" and "cleanly". (I think grammar says that an adverb should be used here, but people say "nice and clean" actually.)

File: man/logcontrol-example.c [hunk 5]
```diff
@@ -0,0 +1,205 @@
...
+   * https://www.freedesktop.org/software/systemd/man/sd_bus_add_object.html
+   * https://dbus.freedesktop.org/doc/dbus-tutorial.html */
+  check(o.log_level, sd_bus_add_object_vtable(bus, NULL,
+                                              "/org/freedesktop/LogControl1",
+                                              "org.freedesktop.LogControl1",
+                                              vtable,
+                                              &o));
+
+  /* By default the service is assigned an ephemeral name. Also add a fixed one,
+   * so that clients know who to call.
```
Comment (keszybz) [inline man/logcontrol-example.c]: whom

File: man/org.freedesktop.LogControl1.xml [hunk 6]
```diff
@@ -116,6 +116,21 @@ node /org/freedesktop/LogControl1 {
...
 
+  <refsect1>
+    <title>Example</title>
+
+    <example>
+      <title>Create a simple listener on the bus that implements LogControl1</title>
+
+      <programlisting><xi:include href="logcontrol-example.c" parse="text"/></programlisting>
+
+      <para>This creates a simple client on the bus (the user bus, when run as normal user). It implements
```
Comment (keszybz) [inline man/org.freedesktop.LogControl1.xml]: This implies the system bus when run as root. But I don't think that's the case. `sd_bus_default()` checks if the slice of the caller is called `user-*.slice`. `user@0.service` runs in `/user.slice/user-0.slice`, so you'd get the user bus too.

The docs for `sd_bus_default()` are pretty bad. I couldn't answer this question based on the man page, so I think we should fix the man page.

**Review by keszybz — CHANGES_REQUESTED** [review]
Looks pretty good. Some grammar nitpicks.

Also, please do run `gcc -Wall man/logcontrol-example.c -lsystemd`. You'll be amazed ;)

**Review by keszybz — APPROVED** [review]
LGTM.

File: man/logcontrol-example.c [hunk 7]
```diff
@@ -0,0 +1,232 @@
...
+    return sd_bus_message_append(reply, "s", log_level_table[o->log_level]);
+
+  if (strcmp(property, "LogTarget") == 0)
+    return sd_bus_message_append(reply, "s", log_target_table[o->log_target]);
+
+  if (strcmp(property, "SyslogIdentifier") == 0)
+    return sd_bus_message_append(reply, "s", o->syslog_identifier);
+
+  return sd_bus_error_setf(error,
+                           SD_BUS_ERROR_INVALID_ARGS,
```
Comment (poettering) [inline man/logcontrol-example.c:107]: SD_BUS_ERROR_UNKNOWN_PROPERTY

File: man/logcontrol-example.c [hunk 8]
```diff
@@ -0,0 +1,232 @@
...
+  /* The bus should be relinquished before the program terminates. The cleanup
+   * attribute allows us to do it nicely and cleanly whenever we exit the
+   * block.
+   */
+  _cleanup_(sd_bus_flush_close_unrefp) sd_bus *bus = NULL;
+
+  /* Acquire a connection to the bus, letting the library work out the details.
+   * https://www.freedesktop.org/software/systemd/man/sd_bus_default.html
+   */
+  sd_bus_default(&bus);
```
Comment (poettering) [inline man/logcontrol-example.c:193]: why not use your check() macro here?

File: man/logcontrol-example.c (hunk 7 above)
Comment (bluca) [reply man/logcontrol-example.c:107]: https://github.com/systemd/systemd/pull/27378

**Pull request merged**